### PR TITLE
Fix warning in event emitter

### PR DESCRIPTION
**Pull request by nunnatsa**
The event emmiter now checks that an interface object parameter is not a
nil interface, but does not check for simple nil. In this case, the
event emmiter writes this warning to the log:

```
Could not construct reference to: '<nil>' due to: 'can't reference a nil
object'. Will not report event...
```

This PR add the simple == nil check back, in addition to the reflection
check.

Signed-off-by: Nahshon Unna-Tsameret <nunnatsa@redhat.com>

**Reviewer Checklist**

> Reviewers are supposed to review the PR for every aspect below one by one. To check an item means the PR is either "OK" or "Not Applicable" in terms of that item. All items are supposed to be checked before merging a PR. 

- [ ] PR Message
- [ ] Commit Messages
- [ ] How to test
- [ ] Unit Tests
- [ ] Functional Tests
- [ ] User Documentation
- [ ] Developer Documentation
- [ ] Upgrade Scenario
- [ ] Uninstallation Scenario
- [ ] Backward Compatibility
- [ ] Troubleshooting Friendly
  
**Release note**:

```release-note
None
```

### CODE REVIEW

**Comment by nunnatsa** [comment]
ignore tiny coverage reduction
/override coverage/coveralls

**Comment by nunnatsa** [comment]
ci/prow/hco-e2e-upgrade-index-awe lane passed

/override ci/prow/hco-e2e-upgrade-index-azure

**Comment by nunnatsa** [comment]
ci/prow/hco-e2e-upgrade-index-sno-aws lane passed

/override ci/prow/hco-e2e-upgrade-index-sno-azure

File: pkg/util/event_emmiter.go [hunk 1]
```diff
@@ -39,13 +39,28 @@ func (ee eventEmitter) EmitEvent(object runtime.Object, eventType, reason, msg s
...
+// a pointer variable that holds nil.
+func IsActuallyNil(object interface{}) bool {
+	if object == nil {
+		return true
+	}
+
+	t := reflect.ValueOf(object)
+
+	switch t.Kind() {
+	case reflect.Chan, reflect.Func, reflect.Map, reflect.Pointer, reflect.UnsafePointer, reflect.Interface, reflect.Slice:
```
Comment (orenc1) [inline pkg/util/event_emmiter.go:61]: what about other reflect kinds such as `Struct` and `Array`? they can be nil as well, no?

File: pkg/util/event_emmiter_test.go [hunk 2]
```diff
@@ -140,6 +147,29 @@ var _ = Describe("", func() {
 			})
 
 		})
+
+		It("should not update resource if it's nil", func() {
+			ee.Init(nil, nil, recorder)
+
+			By("should not emmit event for all three resources", func() {
```
Comment (orenc1) [inline pkg/util/event_emmiter_test.go:154]: typo: emmit --> emit
same as in line 167

Comment (nunnatsa) [reply pkg/util/event_emmiter_test.go:154]: Fixed.

**Review by orenc1 — APPROVED** [review]
/approve

**Comment by nunnatsa** [comment]
Ignore tiny coverage reduce
/override coverage/coveralls

**Pull request merged**